doc/source/ray-air/check-ingest.rst

```diff
@@ -0,0 +1,241 @@
+that you pass to the Trainer. Dataset blocks that don't fit into memory will be spilled to disk. Note that when you create
+the dataset initially, typically only the first block and block metadata is read into memory. The rest of the blocks are
+not loaded until ``fit`` is called.
+
+**Preprocessing**: Next, if a preprocessor is defined, AIR will ``fit`` the preprocessor (e.g., for stateful preprocessors) on the
+``"train"`` dataset, and then ``transform`` all given datasets with the fitted preprocessor. This is done by calling ``prep.fit_transform()``
+on the train dataset passed to the Trainer, followed by ``prep.fit()`` on remaining datasets. Preprocessors use Dataset APIs to execute
+preprocessing in a parallelized way across the cluster. Both read and preprocessing stages use Ray tasks under the hood.
+
+**Training**: Finally, AIR passes a reference to the preprocessed dataset to Train workers (Ray actors) launched by the Trainer. Each worker then
+typically calls ``iter_batches``, ``to_tf``, or ``to_torch`` to iterate over the dataset reference retrieved by ``get_dataset_shard``.
+These read methods load blocks of the dataset into the local worker's memory in a streaming fashion, only fetching / prefetching a
+limited number of blocks at once. Workers loop over the dataset blocks repeatedly until training completes.
+
+Bulk Ingest
+~~~~~~~~~~~
+
+By default, AIR loads all Dataset blocks into the object store at the start of training. This provides the best performance if the
+cluster has enough aggregate memory to fit all the data blocks in object store memory. Note that data often requires more space
+when loaded uncompressed in memory than when resident in storage.
+
+If there is insufficient object store memory, blocks may be spilled to disk during reads or preprocessing. Ray will print log messages
+if spilling is occuring, and you can check this as well with the ``ray memory --stats-only`` utility. If spilling is happening, take
+care to ensure the cluster has enough disk space to handle the spilled blocks. Alternatively, consider using machine with more memory /
+more machines to avoid spilling.
+
+Streamed Ingest
+~~~~~~~~~~~~~~~
+
+This section is a placeholder.
+
+AIR will support streamed ingest and enable it by default by Beta. Streamed ingest is preferable when you are using large
+datasets that don't fit into memory, and also don't need advanced training quality features such as global random shuffle.
+
+Ingest and Ray Tune
+-------------------
+
+.. note::
+
+    Train always uses Tune as the execution backend under the hood, even when running just ``Trainer.fit()`` directly (this is treated as a single-trial experiment). This ensures consistency of execution.
+
+When using ``Ray Tune`` with AIR training jobs, you can choose to either provide an experiment-wide Dataset that is shared among all trials,
+or have per-trial datasets. Trials will always run its preprocessing separately, so Dataset sharing will only deduplicate
```

Comment:
Hmm following up on this I wonder if we should also expose a way to allows users to preprocess datasets outside the Trainer to avoid duplicate work (but still pass in the Preprocessor so it's checkpointed and can be accessed by the predictor)

Reply:
Yeah agreed, though probably should prioritize based on feedback.